packages/graphql/src/schema-model/annotation/AuthorizationAnnotation.ts

```diff
@@ -0,0 +1,129 @@
+
+export const AuthorizationFilterOperationRule = [
+    "READ",
+    "UPDATE",
+    "DELETE",
+    "CREATE_RELATIONSHIP",
+    "DELETE_RELATIONSHIP",
+] as const;
+
+export const AuthorizationValidateOperationRule = [
+    "READ",
+    "CREATE",
+    "UPDATE",
+    "DELETE",
+    "CREATE_RELATIONSHIP",
+    "DELETE_RELATIONSHIP",
+] as const;
+
+export type AuthorizationFilterOperation =
+    (typeof AuthorizationFilterOperationRule)[keyof typeof AuthorizationFilterOperationRule];
+
+export type AuthorizationValidateOperation =
+    (typeof AuthorizationValidateOperationRule)[keyof typeof AuthorizationValidateOperationRule];
+
+export type ValidateWhen = "BEFORE" | "AFTER";
+
+export class AuthorizationAnnotation implements Annotation {
+    name = "AUTHORIZATION";
+    filter?: AuthorizationFilterRule[];
+    validate?: AuthorizationValidateRule[];
+
+    constructor({
+        filter,
+        validate,
+    }: {
+        filter?: AuthorizationFilterRule[];
+        validate?: AuthorizationValidateRule[];
+    }) {
+        this.filter = filter;
+        this.validate = validate;
+    }
+}
+
+export const AuthorizationFilterRuleArguments = {
+    operations: "operations",
+    requireAuthentication: "requireAuthentication",
+    where: "where",
+};
+export type AuthorizationFilterRuleConstructor = {
+    operations?: AuthorizationFilterOperation[];
+    requireAuthentication?: boolean;
+    where: AuthorizationFilterWhere;
+}
+export class AuthorizationFilterRule {
+    operations: AuthorizationFilterOperation[];
+    requireAuthentication: boolean;
+    where: AuthorizationFilterWhere;
+
+    constructor({ operations, requireAuthentication, where }: AuthorizationFilterRuleConstructor) {
+        this.operations = operations ?? [...AuthorizationFilterOperationRule];
+        this.requireAuthentication = requireAuthentication === undefined ? true : requireAuthentication;
+        this.where = new AuthorizationFilterWhere(where);
+    }
+}
+
+export type AuthorizationValidateRuleConstructor = {
+    operations?: AuthorizationValidateOperation[];
+    requireAuthentication?: boolean;
+    where: AuthorizationFilterWhere;
+    when?: ValidateWhen[];
+}
+
+export class AuthorizationValidateRule {
+    operations: AuthorizationValidateOperation[];
+    requireAuthentication: boolean;
+    where: AuthorizationFilterWhere;
+    when: ValidateWhen[];
+
+    constructor({
+        operations,
+        requireAuthentication,
+        where,
+        when,
+    }: AuthorizationValidateRuleConstructor) {
+        this.operations = operations ?? [...AuthorizationValidateOperationRule];
+        this.requireAuthentication = requireAuthentication === undefined ? true : requireAuthentication;
+        this.where = new AuthorizationFilterWhere(where);
+        this.when = when ?? ["BEFORE", "AFTER"];
+    }
+}
+
+export const AuthorizationFilterRuleWhereArguments = {
+    jwtPayload: "jwtPayload",
+    node: "node",
+};
+export class AuthorizationFilterWhere {
```

Comment:
I would change the naming of this as to not to conflict with filter rules.

```suggestion
export class AuthorizationWhere {
```